pandas/tests/indexes/common.py

```diff
@@ -278,6 +284,8 @@ def test_ensure_copied_data(self):
 
             index_type = index.__class__
             result = index_type(index.values, copy=True, **init_kwargs)
+            result = self._fix_tz(result, index)
```

Comment:
you don't need to touch this because ``_holder`` will also be different.

We shold prob refactor this ``init_kwargs`` bizness to instead be a class method, that way each class can construct things easily

IOW

```
class .....:

     # e.g. for PeriodIndex
     _init_kwargs = lambda index: return {'freq': index.freq'}

     # in the base class (no-op)
     _init_kwargs = lambda index: return {}
```

Reply:
`_holder` here will also be `DatetimeIndex`, which isn't sufficient because `DatetimeIndex` constructor doesn't have TZ information. The problem is that the constructor `DatetimeIndex(index.values, ...)` is getting a numpy datetime64 array of *tz-naive* timestamps. There currently doesn't seem to be an argument to `DatetimeIndex`'s constructor to tell it what timezone to convert to presuming that the input data is UTC (the `tz=` argument localizes to the specified timezone), so there's not currently a way to do this through `init_kwargs`. Hence the need currently to call `_fix_tz` to first localize to UTC and then convert to the TZ of `create_index()`. 

I can do the suggested refactor though.

Reply:
yes prob need to refactor a bit,